File: Chapter 04/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter04/section_4_4_anti_patterns/after_rich_domain_thin_controller/businesslogic/OrderRequest.java
package com.grokkingsoftwarearchitecture.chapter04.section_4_4_anti_patterns.after_rich_domain_thin_controller.businesslogic;

import com.grokkingsoftwarearchitecture.chapter04.section_4_4_anti_patterns.after_rich_domain_thin_controller.domainmodels.Item;
import java.util.List;

/**
 * DTO (Data Transfer Object) for incoming requests.
 */
public class OrderRequest {
    public int customerId;
    public List<Item> items;
}

File: Chapter 04/4.4 Anti-Patterns/After-RichDomain/After/After.BusinessLogic/OrderRequest.cs
using After.DomainModels;

namespace After.BusinessLogic
{
    // DTO (Data Transfer Object) for incoming requests    
    public class OrderRequest
    {
        public int CustomerId { get; set; }
        public List<Item> Items { get; set; }
    }

    // The Business Layer owns its own interface
    public interface IOrderService
    {
        int CreateOrder(OrderRequest request);
    }
}